### PR TITLE
Feat: Add KMS decrypt entrypoint.

### PR DESCRIPTION
**Pull request by steveworley**
# Checklist
- [ ] Affected Issues have been mentioned in the Closing issues section
- [ ] Documentation has been written/updated.
- [ ] Changelog entry has been written

Explain the **details** for making this change. What existing problem does the pull request solve?

# Changelog Entry

# Closing issues
Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

File: images/bay-php/entrypoints/bay-shared-temp-files-dir-check.sh [hunk 1]
```diff
@@ -1,11 +1,12 @@
 #!/bin/sh
-set -euo pipefail
+# set -euo pipefail
```
Comment (nicksantamaria) [inline images/bay-php/entrypoints/bay-shared-temp-files-dir-check.sh:2]: What is this commented out for @steveworley ?

Comment (nicksantamaria) [reply images/bay-php/entrypoints/bay-shared-temp-files-dir-check.sh:2]: aaaaaaaand I found out why.

The way lagoon invokes entrypoint scripts means any bash options you change persist in other scripts. In particular, the `-u` flag breaks a bunch of lagoon commons scripts that indiscriminately use undefined variables.

**Review by nicksantamaria — APPROVED** [review]

File: images/bay-php/entrypoints/100-kms-decrypt.sh [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+expr "$*" : ".*--help" > /dev/null && usage
+
+echoerr() { printf "%s\n" "$*" >&2 ; }
+info()    { echoerr "[INFO]    $*" ; }
+warning() { echoerr "[WARNING] $*" ; }
+error()   { echoerr "[ERROR]   $*" ; }
+fatal()   { echoerr "[FATAL]   $*" ; exit 1 ; }
+
+info "decrypting files"
+if (find /app/keys -name "*.asc" | grep asc) > /dev/null 2>&1 && [ -n "$AWS_ACCESS_KEY_ID" ] && [ -n "$AWS_SECRET_ACCESS_KEY" ]; then
```
Comment (nicksantamaria) [inline images/bay-php/entrypoints/100-kms-decrypt.sh]: ` (find /app/keys -name "*.asc" | grep asc)` is just a janky way to test if `.asc` files exist in a multi tier directory structure.

File: images/bay-php/entrypoints/100-kms-decrypt.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+if [ -n "$encrypted_files" ] && [ -n "$AWS_ACCESS_KEY_ID" ] && [ -n "$AWS_SECRET_ACCESS_KEY" ]; then
+    for file in $encrypted_files; do
+        info " - ${file} > ${file%.asc}"
+        bay kms decrypt < "${file}" > "${file%.asc}" || error "unable to decrypt ${file}"
+    done
+else
+    info "no files to decrypt"
+fi
+
+# Set optiosn back to previous state.
```
Comment (steveworley) [inline images/bay-php/entrypoints/100-kms-decrypt.sh]:
```suggestion
# Set options back to previous state.
```

File: images/bay-php/entrypoints/bay-shared-temp-files-dir-check.sh (hunk 1 above)
Comment (steveworley) [reply images/bay-php/entrypoints/bay-shared-temp-files-dir-check.sh:2]: Yeah, that's why — I thought I did put a comment to that effect somewhere. We can probably toggle these options like the kms decrypt script is doing, but I wasn't sure exactly for this one.

File: images/bay-php/entrypoints/100-kms-decrypt.sh [hunk 4]
```diff
@@ -20,12 +20,19 @@ error()   { echoerr "[ERROR]   $*" ; }
 fatal()   { echoerr "[FATAL]   $*" ; exit 1 ; }
 
 info "decrypting files"
-if (find /app/keys -name "*.asc" | grep asc) > /dev/null 2>&1 && [ -n "$AWS_ACCESS_KEY_ID" ] && [ -n "$AWS_SECRET_ACCESS_KEY" ]; then
-    files=$(find /app/keys -name "*.asc")
-    for file in $files; do
+
+encrypted_files=""
+if [ -d "/app/keys" ]; then
+    encrypted_files=$(find /app/keys -type f -name "*.asc" -print0 2>/dev/null)
```
Comment (steveworley) [inline images/bay-php/entrypoints/100-kms-decrypt.sh:26]: Testing this:
```
Attaching to bay-cli-1
bay-cli-1  | [INFO]    decrypting files
bay-cli-1  | [INFO]     - /app/keys/another/dir/key.asc > /app/keys/another/dir/key
```
Successfully picked up and decrypted the keys, not that we would advocate for nesting keys like this.

**Review by nicksantamaria — APPROVED** [review]

**Pull request merged**